build_system/src/prepare.rs

```diff
@@ -1,12 +1,66 @@
+use std::ffi::OsStr;
 use std::fs;
 use std::path::{Path, PathBuf};
+use std::process::Output;
 
 use crate::rustc_info::get_rustc_path;
 use crate::utils::{
     cargo_install, create_dir, get_sysroot_dir, git_clone_root_dir, remove_file, run_command,
     run_command_with_output, walk_dir,
 };
 
+fn run_git_command(
+    command: &dyn AsRef<OsStr>,
+    input: &[&dyn AsRef<OsStr>],
+    cwd: Option<&Path>,
+) -> Result<Output, String> {
+    // This is needed on systems where nothing is configured.
+    // git really needs something here, or it will fail.
+    // Even using --author is not enough.
+    let git_cmd = &[
+        &"git" as &dyn AsRef<OsStr>,
+        &"-c" as &dyn AsRef<OsStr>,
+        &"user.name=None" as &dyn AsRef<OsStr>,
+        &"-c" as &dyn AsRef<OsStr>,
+        &"user.email=none@example.com" as &dyn AsRef<OsStr>,
+        &"-c" as &dyn AsRef<OsStr>,
+        &"core.autocrlf=false" as &dyn AsRef<OsStr>,
+        &"-c" as &dyn AsRef<OsStr>,
+        &"commit.gpgSign=false" as &dyn AsRef<OsStr>,
+    ]
+    .into_iter()
+    .chain([command])
+    .chain(input.iter().cloned())
+    .collect::<Vec<_>>()[..];
+    run_command(git_cmd, cwd)
+}
+
+fn run_git_command_with_output(
+    command: &dyn AsRef<OsStr>,
+    input: &[&dyn AsRef<OsStr>],
+    cwd: Option<&Path>,
+) -> Result<(), String> {
+    // This is needed on systems where nothing is configured.
+    // git really needs something here, or it will fail.
+    // Even using --author is not enough.
+    let git_cmd = &[
+        &"git" as &dyn AsRef<OsStr>,
+        &"-c" as &dyn AsRef<OsStr>,
+        &"user.name=None" as &dyn AsRef<OsStr>,
+        &"-c" as &dyn AsRef<OsStr>,
+        &"user.email=none@example.com" as &dyn AsRef<OsStr>,
+        &"-c" as &dyn AsRef<OsStr>,
+        &"core.autocrlf=false" as &dyn AsRef<OsStr>,
+        &"-c" as &dyn AsRef<OsStr>,
+        &"commit.gpgSign=false" as &dyn AsRef<OsStr>,
+    ]
+    .into_iter()
+    .chain([command])
+    .chain(input.iter().cloned())
+    .collect::<Vec<_>>()[..];
+    run_command_with_output(git_cmd, cwd)
+}
```

Comment:
There's some duplication between these 2 functions.
Could you please refactor these functions to avoid this?

Reply:
Moved the command definitions to a const